src/features/web-compat.js

```diff
@@ -234,6 +240,46 @@ export default class WebCompat extends ContentFeature {
         })
     }
 
+    screenLockFix () {
+        // @ts-expect-error Property 'lock' does not exist on type 'ScreenOrientation'
+        window.ScreenOrientation.prototype.lock = new Proxy(async (requestedOrientation) => {
```

Comment:
Asked this in Asana, but asking here as well: is this the correct way to override? Also, how should we check if we need to override since lock/unlock are actually defined?

Reply:
Answered in Asana for the check part: https://app.asana.com/0/0/1206174768741464/1206228892953010/f

As this is a propertyDescriptor I think you want to use `this.wrapProperty` and match the properties returned by:

```
Object.getOwnPropertyDescriptors(window.ScreenOrientation.prototype).lock
> Object { value: lock(), writable: true, enumerable: true, configurable: true }
```

I also scoped a project to make this simpler to understand: https://app.asana.com/0/1163321984198618/1206229471617467/f

Reply:
Got it, thanks a lot for updating the docs!